scio-core/src/main/scala/com/spotify/scio/util/ScioUtil.scala

```diff
@@ -86,8 +91,11 @@ private[scio] object ScioUtil {
     }
   }
 
-  def pathWithShards(path: String): String =
-    path.replaceAll("\\/+$", "") + "/part"
+  def pathWithPrefix(path: String, prefix: String = null): String = {
+    val filePrefix = Option(prefix).getOrElse("part")
+    val stripped = StringUtils.stripEnd(path, "/")
+    s"${stripped}/${filePrefix}"
+  }
```

Comment:
I'm a bit confused by this signature. not giving a `prefix` will end up building with `part`.

I think this could be more logical:
```suggestion
  def pathFolder(path: String): String = pathWithPrefix(path, "")
  def pathWithPrefix(path: String, prefix: String = "part"): String = {
    val stripped = StringUtils.stripEnd(path, "/")
    s"${stripped}/${prefix}"
  }
```
We'll however have to change all occurrences of
```diff
-pathWithPrefix(path, "")
+pathFolder(path)
```
And where `null` is unchecked (I think only one place)
```diff
-pathWithPrefix(path, prefix)
+Option(prefix) match {
+  case Some(p) => pathWithPrefix(path, p)
+  case None => pathWithPrefix(path)
+}
```
To ensure that even with null, a prefix is added.
We could also work with options, with default to `Some("part")`:
```diff
-pathWithPrefix(path, "")
+pathWithPrefix(path, None)
```
and
```diff
-pathWithPrefix(path, prefix)
+Option(prefix) match {
+  case p: Some => pathWithPrefix(path, p)
+  case None => pathWithPrefix(path) // use default prefix
+}
```

Reply:
The original variant always appended `part` so this was some attempt at maintaining backwards compatibility, but you're right that it'd be better to have a clearer api here, will address.